subsystems/mvdm/ntvdm/dos/dos32krnl/process.c

```diff
@@ -796,10 +796,45 @@ WORD DosCreateProcess(IN LPCSTR ProgramName,
         /* Those are handled by NTVDM */
         case SCS_WOW_BINARY:
         {
-            DisplayMessage(L"Trying to load '%S'.\n"
-                           L"WOW16 applications are not supported internally by NTVDM at the moment.\n"
-                           L"Press 'OK' to continue.",
-                           ProgramName);
+            static const PCSTR AppName = "\"%ProgramFiles%\\otvdm\\otvdmw.exe\" ";
+
+            STARTUPINFOA si;
+            PROCESS_INFORMATION pi;
+            CHAR ExpName[MAX_PATH];
+
+            ExpandEnvironmentStringsA(AppName, ExpName, ARRAYSIZE(ExpName) - 1);
+            strcat(ExpName, "\"");         // Add double-quote before ProgramName
+            strcat(ExpName, ProgramName);  // Append Program name
+            strcat(ExpName, "\"");         // Add double-quote after ProgramName
+
+            ZeroMemory(&pi, sizeof(pi));
+            ZeroMemory(&si, sizeof(si));
+            si.cb = sizeof(si);
+
+            /* Create the process */
+            if (CreateProcessA(NULL,    // No Application Name
+                               ExpName, // Just our Command Line
+                               NULL,    // Cannot inherit Process Handle
+                               NULL,    // Cannot inherit Thread Handle
+                               FALSE,   // No handle inheritance
+                               0,       // No extra creation flags
+                               NULL,    // No environment block
+                               NULL,    // No starting directory 
+                               &si,
+                               &pi))
+            {
+                /* Close the handles */
+                CloseHandle(pi.hThread);
+                CloseHandle(pi.hProcess);
+                break;
+            }
+            else
+            {
+                DisplayMessage(L"Trying to load '%S'.\n"
+                               L"WOW16 applications are not supported internally by NTVDM at the moment.\n"
+                               L"Consider installing otvdm in 'C:\\Program files'. Press 'OK' to continue.",
+                               ProgramName);
```

Comment:
A suggestion for the (rare) cases where "Program Files" is localized (the correct name could be simply retrieved after expansion of the corresponding environment variable):
```suggestion
                /* Retrieve the actual path to the "Program Files" directory for displaying the error */
                ExpandEnvironmentStringsA("%ProgramFiles%", ExpName, ARRAYSIZE(ExpName) - 1);

                DisplayMessage(L"Trying to load '%S'.\n"
                               L"WOW16 applications are not supported internally by NTVDM at the moment.\n"
                               L"Consider installing OTVDM in '%S'. Press 'OK' to continue.",
                               ProgramName, ExpName);
```